ui/src/app/components/help-sidebar-icons.tsx

```diff
@@ -113,8 +127,58 @@ export interface IconConfig {
   hasContent: boolean;
 }
 
+export interface StatusIconProps {
+  iconPath: string;
+  iconConfig: IconConfig;
+  config: ConfigResponse;
+  children: React.ReactNode;
+}
+
+export const StatusIcon = ({
+  iconPath,
+  iconConfig,
+  config,
+  children,
+}: StatusIconProps) => {
+  const iconStyle: CSSProperties = showAppsPanel(config)
+    ? { width: '36px', position: 'absolute' }
+    : { width: '22px', position: 'absolute' };
+
+  return (
+    <FlexRow style={containerStyle}>
+      <img
+        src={iconPath}
+        alt={iconConfig.label}
+        style={iconStyle}
+        data-test-id={'help-sidebar-icon-' + iconConfig.id}
+      />
+      {children}
+    </FlexRow>
+  );
+};
+
+const displayAppStatusIcon = (
+  iconConfig: IconConfig,
+  workspaceNamespace: string,
+  userSuspended: boolean,
+  config: ConfigResponse,
+  status: AppStatus,
+  appType: UIAppType
+) => {
+  const appTypeAssets = appAssets.find((aa) => aa.appType === appType);
+  return (
+    <StatusIcon {...{ config, iconConfig }} iconPath={appTypeAssets?.icon}>
+      <AppStatusIcon
```

Comment:
I find the dual use of "status icon" confusing here. It's being used for both the Cromwell/Jupyter logo and for the small icon displaying the status. Could one of those uses be renamed?

Reply:
I think that's a fair point. I ended up renaming both components:

- AppStatusIcon and RuntimeStatusIcon are now AppStatusIndicator and RuntimeStatusIndicator.
- StatusIcon has been renamed to CompoundIcon. I named it this, because its purpose is to combine an icon file (a picture of a pig or the Jupyter logo) with something else (currently an indicator representing status). I am up for alternate names, but I think that at least clears up some of the confusion.

Reply:
Sounds good to me!